### PR TITLE
swipe-to-open: Adding swipe to open drawer with custom options

### PR DESCRIPTION
**Pull request by briblanch**
Description
---
- Added swipe to open drawer with custom options.

What Was Changed
---
- Added pan gesture with logic to handle the swipe to open and close the drawer.
- Added new options for the `WSideMenuVC`:
    - `swipeToOpen`: The one flag that needs to be enabled to allow the user to swipe to open the drawer
    - `swipeToOpenThreshold`: The percentage of the left side of the view used that can be swiped on and the drawer will open. Defaults to the left third.
    - `autoOpenThreshold`: The percentage the drawer has to be swiped before it will automatically open. Defaults to halfway open.

Testing
---
1. Run the sample app and swiping to open the drawer from anywhere should work.
* Swiping right on the left third of the main view will start to open the drawer.
* Releasing before the drawer reaches half way should close the drawer.
* Releasing after the drawer reaches half way should open the drawer.
* Swiping the left when the drawer is closed should do nothing.
* Swiping anywhere outside the drawer when it is open should start to close it.

---

Please Review: @Workiva/mobile

### CODE REVIEW

**Comment by jordanross-wf** [comment]
When swiping open the drawer, the shadow will not appear until you let go, but once the shadow is there, it stays until you close the drawer. This should be more consistent by having the shadow appear as the user starts opening the drawer.

**Comment by brianblanchard-wf** [comment]
@jordanross-wf I added the swipe to add an animation to the shadow of the drawer as well when it comes out.

**Comment by briblanch** [comment]
@jordanross-wf I added the shadow to animate as the drawer is pulled in and out.

File: Source/WSideMenuVC.swift [hunk 1]
```diff
@@ -196,6 +257,16 @@ public class WSideMenuVC: WSizeVC {
             closeSideMenu()
         }
     }
+    
+    public func hideStatusBar(hide: Bool) {
+        statusBarHidden = hide
+        
+        if (hide) {
```
Comment (jeffscaturro-wf) [inline Source/WSideMenuVC.swift]: this could just be a one-liner `UIApplication.sharedApplication().delegate?.window!!.windowLevel = hide ? UIWindowLevelStatusBar : UIWindowLevelNormal`.

Also, are we force unwrapping `window` here?

**Comment by jordanross-wf** [comment]
+10, works as expected and UX approval received verbally.

- [ ] Appropriate reviews.
- [ ] Unit test created or updated.
- [ ] Acceptance criteria met.
- [ ] Edge cases tested.
- [ ] Rosie has run.

Merging into `master`.
